pandas/core/resample.py

```diff
@@ -1583,25 +1603,46 @@ def _take_new_index(obj, indexer, new_index, axis=0):
 
 
 def _get_range_edges(first, last, offset, closed='left', base=0):
+    if type(first) != type(last):
```

Comment:
I think its better if you just create a new `_get_period_range_edges` function with the period logic and call the original function `_get_datetime_range_edges`. The function is looking unwieldy.

Reply:
agreed and fixed.